staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -1323,16 +1346,18 @@ func (r *Request) transformResponse(ctx context.Context, resp *http.Response, re
 			// 2. Apiserver sends back the headers and then part of the body
 			// 3. Apiserver closes connection.
 			// 4. client-go should catch this and return an error.
-			klog.V(2).Infof("Stream error %#v when reading response body, may be caused by closed connection.", err)
+			klog.FromContext(ctx).V(2).Info("Stream error when reading response body, may be caused by closed connection", "err", err)
```

Comment:
That would then do a `klog.FromContext` in all cases, whether there was an error or not. It's better to call it only if really needed, even if that means writing `klog.FromContext` twice.

Reply:
Honestly, looking at the code it will invoke `klog.FromContext` in almost all paths, so I'm not sure I follow your argument. Also, I've noticed that `logBody` accepts context, which imo is a bit of an overkill, it should just accept logger as well. But that's not blocking this PR, can be handled some other time.

Reply:
You are right, I missed that this is the "normal" path. Changed.

> Also, I've noticed that logBody accepts context, which imo is a bit of an overkill

"Overkill" in the sense of "doesn't need it", not as in the sense of "performance". It's mostly for convenience: most callers don't have a logger, so if we were to require a logger, we'd end up with more `klog.FromContext` calls scattered in various places.

But if we have a logger, then passing it indeed saves some overhead for another such call. That makes sense to me, so I changed it.

> But that's not blocking this PR, can be handled some other time.

Let's do it right in this PR, such things tend to get forgotten:
https://github.com/kubernetes/kubernetes/compare/ae80977df9965a291f8634ac6b82c176f2813c88..7b6f0dfea045463da617ae00e85d3c31359c18ee